app/controllers/courses_controller.rb

```diff
@@ -28,6 +28,21 @@ def self.registered(app)
 
             @course_coll = app.settings.courses_db.collection("courses#{params[:semester]}")
             @section_coll = app.settings.courses_db.collection("sections#{params[:semester]}")
+            @prof_coll = app.settings.courses_db.collection("profs#{params[:semester]}")
+          end
+
+
+          # Route for professors, use 'name' or 'course' to filter
+          app.get '/v0/courses/professors' do
```

Comment:
Do you think the new endpoint should still be separated by semester? As in, `v0/professors` just returns professors for the current semester? Another idea is returning all professors, but each with a hash property of `{ semester => array of courses }`. Though that wouldn't be very consistent with the rest of the API.